smart-contracts/contracts/cl-vault/src/vault/admin.rs

```diff
@@ -234,67 +187,67 @@ mod tests {
         assert_eq!((), verify_resp);
     }
 
-    #[test]
-    fn test_execute_claim_strategist_rewards_success() {
-        let treasury = Addr::unchecked("bob");
-        let mut deps = mock_dependencies();
-        let rewards = vec![coin(12304151, "uosmo"), coin(5415123, "uatom")];
-        STRATEGIST_REWARDS
-            .save(
-                deps.as_mut().storage,
-                &CoinList::from_coins(rewards.clone()),
-            )
-            .unwrap();
-
-        VAULT_CONFIG
-            .save(
-                deps.as_mut().storage,
-                &VaultConfig {
-                    performance_fee: Decimal::percent(20),
-                    treasury: treasury.clone(),
-                    swap_max_slippage: Decimal::percent(10),
-                    dex_router: Addr::unchecked("bob-router"),
-                },
-            )
-            .unwrap();
-
-        let response =
-            execute_claim_strategist_rewards(deps.as_mut(), mock_info(treasury.as_str(), &[]))
-                .unwrap();
-        assert_eq!(
-            CosmosMsg::Bank(BankMsg::Send {
-                to_address: treasury.to_string(),
-                amount: sort_tokens(rewards)
-            }),
-            response.messages[0].msg
-        )
-    }
-
-    #[test]
-    fn test_execute_claim_strategist_rewards_not_admin() {
-        let treasury = Addr::unchecked("bob");
-        let mut deps = mock_dependencies();
-        let rewards = vec![coin(12304151, "uosmo"), coin(5415123, "uatom")];
-        STRATEGIST_REWARDS
-            .save(deps.as_mut().storage, &CoinList::from_coins(rewards))
-            .unwrap();
-
-        VAULT_CONFIG
-            .save(
-                deps.as_mut().storage,
-                &VaultConfig {
-                    performance_fee: Decimal::percent(20),
-                    treasury: treasury.clone(),
-                    swap_max_slippage: Decimal::percent(10),
-                    dex_router: Addr::unchecked("bob-router"),
-                },
-            )
-            .unwrap();
-
-        let err =
-            execute_claim_strategist_rewards(deps.as_mut(), mock_info("alice", &[])).unwrap_err();
-        assert_eq!(ContractError::Unauthorized {}, err)
-    }
+    // #[test]
```

Comment:
With what test did we replace this commented part?

Reply:
This should be covered in two parts:

During the migration with pending rewards, we are clearing the state after sending any pending strategist rewards to be collected during the migration itself. This should be covered by the `contract.rs::test_migrate_with_rewards_execute_steps` test case, but it only partially covered, we are asserting only the clear but not the send. I will add the mock/assertion for that.

Additionally, it should be asserted in a slightly different way against the `execute_collect_rewards` method, which now sends the rewards directly instead of saving them as a state item and keeping them as contract balance. This is already covered by the `test_tube::rewards.rs::test_collect_rewards_with_rewards_works` test case.

Reply:
Done

Reply:
Looks good, thanks!